### PR TITLE
Filtrers ready, update attributes of car model and filter near of you ready

### PR DESCRIPTION
**Pull request by IsmaelMontiel710**
# [Feature] Enhanced Car Entity Model, Filter Options, and Detailed View Improvements
#61 #59 #49 
## Summary

This Pull Request enhances the `CarEntity` model by adding fields for listing dates and detailed car attributes, improves filter functionality, and expands the detailed view with more car attributes.

## Changes Made

1. **Updated `CarEntity` Model:**
   - Added fields:
      - **`listingDate`**: Records when the listing was created.
      - **`lastUpdated`**: Indicates the last update date for the listing.
   - Additional attributes to enhance details for each car, such as `features`, `location`, and `vin` (Vehicle Identification Number).

2. **Expanded Filter Options:**
   - **Location Filter ("Near You")**: Allows users to filter car listings based on geographic proximity.
   - **Condition Filter**: Enables filtering by car condition, such as new or used.
   - **Dynamic Brand Search**: Implemented an `AutoCompleteTextView` to allow users to search and filter brands dynamically.

3. **Detailed View Improvements:**
   - Enhanced the car details screen to display the following additional attributes:
      - Transmission, Fuel Type, Engine Capacity, Condition, VIN, and Previous Owners.
   - Optimized layout to accommodate new fields while ensuring readability.

4. **Enhanced User Experience:**
   - Improved filter reset functionality to allow users to quickly clear all filters and reload the complete list.
   - Added formatting for price and mileage input fields to improve usability.

## Purpose

These changes aim to:
- Provide a more comprehensive and detailed view of each car listing.
- Allow users to refine searches with advanced filtering options for an optimized browsing experience.
- Enable location-based filtering, helping users find cars near their location.

## Testing

- Validated model updates and ensured new fields are correctly populated.
- Verified filter options for correct data output based on user input.
- Conducted usability testing on the detailed view to ensure all new fields display as expected.

Please review and provide feedback. Thank you for considering these updates to the CarHive application!
image
image
image
image
image

### CODE REVIEW

**Review by Uli19 — APPROVED** [review]
What a cool engineer we have, good job

**Review by Kamila04 — APPROVED** [review]
Good job 👍

File: app/src/main/res/drawable/button_more_info_background.xml
Comment (Gavalencia12) [inline app/src/main/res/drawable/button_more_info_background.xml:1]: The comments in English do not currently affect the application.

**Review by Gavalencia12 — APPROVED** [review]
The comments in English do not currently affect the application.

**Review by Angel591Diaz — APPROVED** [review]

File: app/src/main/res/drawable/button_more_info_background.xml
Comment (IsmaelMontiel710) [reply app/src/main/res/drawable/button_more_info_background.xml:1]: ?????

**Pull request merged**